### ISSUE TITLE
Waiting for DataSourceInitializer to run schema.sql

### ISSUE THREAD

**Issue by candrews** [lukas-krecan/ShedLock#121]
I'm using Spring Boot 2.1.2 and Shedlock 2.2.1.

I'm using schema.sql (as documented at https://docs.spring.io/spring-boot/docs/current/reference/html/howto-database-initialization.html#howto-initialize-a-database-using-spring-jdbc ) to create the schema used by shedlock. When using `@EnableSchedulerLock` Spring Boot isn't running `src/main/resources/schema.sql` as it should resulting in Spring Boot failing to start up.

Here's SchedulingConfig.java:
```
import javax.sql.DataSource;

import net.javacrumbs.shedlock.core.LockProvider;
import net.javacrumbs.shedlock.provider.jdbctemplate.JdbcTemplateLockProvider;
import net.javacrumbs.shedlock.spring.annotation.EnableSchedulerLock;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableScheduling;

/** Scheduling configuration and scheduled tasks.
 * @author Craig Andrews
 *
 */
@EnableScheduling
@Configuration
@EnableSchedulerLock(defaultLockAtMostFor = "PT10M")
public class SchedulingConfig {
	@Value("${schedlock.schema}")
	private String schedLockSchema;

	@Bean
	public LockProvider lockProvider(final DataSource dataSource) {
		return new JdbcTemplateLockProvider(dataSource, schedLockSchema + ".shedlock");
	}

}
```

Here's src/main/resources/schema.sql:
```
BEGIN;
CREATE SCHEMA IF NOT EXISTS myapp;
COMMIT;
```

And here's the startup output (Liquibase is the first error as the schema it uses is created by schema.sql, and that isn't happening):
```
2019-01-30 17:04:26.463  INFO 3045 --- [           main] c.i.t.IndexControllerTest                : No active profile set, falling back to default profiles: default
2019-01-30 17:04:26.766  INFO 3045 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data repositories in DEFAULT mode.
2019-01-30 17:04:26.783  INFO 3045 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 17ms. Found 2 repository interfaces.
2019-01-30 17:04:26.886  INFO 3045 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'schedulingConfig' of type [com.integralblue.myapp.SchedulingConfig$$EnhancerBySpringCGLIB$$a45017c8] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2019-01-30 17:04:26.889  INFO 3045 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.boot.autoconfigure.jdbc.DataSourceConfiguration$Hikari' of type [org.springframework.boot.autoconfigure.jdbc.DataSourceConfiguration$Hikari] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2019-01-30 17:04:26.899  INFO 3045 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'spring.datasource-org.springframework.boot.autoconfigure.jdbc.DataSourceProperties' of type [org.springframework.boot.autoconfigure.jdbc.DataSourceProperties] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2019-01-30 17:04:26.913  INFO 3045 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'dataSource' of type [com.zaxxer.hikari.HikariDataSource] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2019-01-30 17:04:26.916  INFO 3045 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'lockProvider' of type [net.javacrumbs.shedlock.provider.jdbctemplate.JdbcTemplateLockProvider] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2019-01-30 17:04:26.988  INFO 3045 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration' of type [org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration$$EnhancerBySpringCGLIB$$dc5eaadf] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2019-01-30 17:04:27.052  INFO 3045 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-4 - Starting...
2019-01-30 17:04:27.053  INFO 3045 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-4 - Start completed.
2019-01-30 17:04:27.200  INFO 3045 --- [           main] liquibase.executor.jvm.JdbcExecutor      : SELECT COUNT(*) FROM MYAPP.DATABASECHANGELOGLOCK
2019-01-30 17:04:27.201  INFO 3045 --- [           main] liquibase.executor.jvm.JdbcExecutor      : CREATE TABLE MYAPP.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID))
2019-01-30 17:04:27.202  WARN 3045 --- [           main] o.s.w.c.s.GenericWebApplicationContext   : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LockException: liquibase.exception.DatabaseException: Schema "MYAPP" not found; SQL statement:
CREATE TABLE MYAPP.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID)) [90079-197] [Failed SQL: CREATE TABLE MYAPP.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID))]
2019-01-30 17:04:27.202  INFO 3045 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-4 - Shutdown initiated...
2019-01-30 17:04:27.205  INFO 3045 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-4 - Shutdown completed.
2019-01-30 17:04:27.213  INFO 3045 --- [           main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2019-01-30 17:04:27.214 ERROR 3045 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.LockException: liquibase.exception.DatabaseException: Schema "MYAPP" not found; SQL statement:
CREATE TABLE MYAPP.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID)) [90079-197] [Failed SQL: CREATE TABLE MYAPP.DATABASECHANGELOGLOCK (ID INT NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TIMESTAMP, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID))]
``

By removing the `@EnableSchedulerLock`, everything starts up and runs fine without error.

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. It looks similar to https://github.com/lukas-krecan/ShedLock/issues/118 (please check if the resolution would not work for you). Could you please try to create an example reproducing the bug. I will try to reproduce it, but there is always some important detail missing in the bug description so I usually can't reproduce it myself.

**Comment by candrews** [comment]
I don't see a resolution in that issue... I'm not using any code I developed myself. I'm relying on the [data source initializer functionality built into Spring Boot](https://docs.spring.io/spring-boot/docs/current/reference/html/howto-database-initialization.html#howto-initialize-a-database-using-spring-jdbc). I can't change it to use `@PostConstruct` and I doubt the Spring Boot people are eager to do so either :)

**Comment by lukas-krecan** [comment]
Here is my attempt to reproduce the issue https://github.com/lukas-krecan/ShedLock/pull/124/files Spring Boot, schema.sql, liquibase, `@EnableSchedulerLock` everything works as expected. Can you please look at the example and tell my what am I missing?

**Comment by lukas-krecan** [comment]
I am able to reproduce the bug now.

**Comment by lukas-krecan** [comment]
fixed in 2.3.0

**Comment by candrews** [comment]
Thank you!
